zinc/src/main/scala/sbt/internal/inc/javac/AnalyzingJavaCompiler.scala

```diff
@@ -35,11 +38,13 @@ final class AnalyzingJavaCompiler private[sbt] (
    * @param options  The options for the Java compiler
    * @param output   The output configuration for this compiler
    * @param callback  A callback to report discovered source/binary dependencies on.
+   * @param classfileManager The component that manages generated class files.
    * @param reporter  A reporter where semantic compiler failures can be reported.
    * @param log       A place where we can log debugging/error messages.
    * @param progressOpt An optional compilation progress reporter.  Where we can report back what files we're currently compiling.
    */
-  def compile(sources: Seq[File], options: Seq[String], output: Output, callback: AnalysisCallback, reporter: Reporter, log: Logger, progressOpt: Option[CompileProgress]): Unit = {
+  def compile(sources: Seq[File], options: Seq[String], output: Output, callback: AnalysisCallback,
+    classfileManager: ClassFileManager, reporter: Reporter, log: Logger, progressOpt: Option[CompileProgress]): Unit = {
```

Comment:
iiuc `classfileManager` lives within the scope of each level of compile: `doCompile` will start with an empty set generated of files and collect them.

https://github.com/sbt/zinc/blob/4bc93500d2ce98da15f61fdc291aacceee4f440e/internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala#L76-L78